src/qibolab/platforms/platform.py

```diff
@@ -85,3 +102,51 @@ def set_bias(self, qubit, bias):
 
     def get_bias(self, qubit):
         return self.qubits[qubit].flux.bias
+
+
+class AcquisitionType(Enum):
+    """
+    Types of data acquisition from hardware.
+
+    SPECTROSCOPY: Zurich Integration mode for RO frequency sweeps,
+    INTEGRATION: Demodulate and integrate the waveform,
+    RAW: Acquire the waveform as it is,
+    DISCRIMINATION: Demodulate, integrate the waveform and discriminate among states based on the voltages
+
+    """
+
+    RAW = auto()
+    INTEGRATION = auto()
+    DISCRIMINATION = auto()
+
+
+class AveragingMode(Enum):
+    """
+    Types of data averaging from hardware.
+
+    CYLIC: Better averaging for noise,
+    SINGLESHOT: False averaging,
+    [SEQUENTIAL: Worse averaging for noise]
+
+    """
+
+    CYCLIC = auto()
+    SINGLESHOT = auto()
+
+
+@dataclass
```

Comment:
If it is created, but not supposed to be modified (once constructed it should be read-only, since they are input parameters - more correctly called *immutable*), consider to mark it as `frozen`, since it gives further guarantees on the instances' behavior.

```suggestion
@dataclass(frozen=True)
```

Reply:
I will change them inside of the driver, I don't now if this is ok but I will read them and subtitute them accordingly by the parameter on my driver that matches that behavior. In this case may not be very clear as zurich parameters are named in the same way.

```
if options.averaging_mode is AveragingMode.CYCLIC:
    options.averaging_mode = lo.AveragingMode.CYCLIC
elif options.averaging_mode is AveragingMode.SINGLESHOT:
    options.averaging_mode = lo.AveragingMode.SINGLE_SHOT
```

That way I can pass options to all the functions that may need to read instead of self.averaging_mode on the driver.

Reply:
Okay, I will work with a copy inside of the driver because they are different types.